### ISSUE TITLE
Can I use stream training with LSTM network?

### ISSUE THREAD

**Issue by fe-dox** [BrainJS/brain.js#229]
## *What* is wrong?
I'm not sure am I able to use Training stream while training LTSM network.
When I try to do that i get this error:

```
var trainStream = net.createTrainStream({
                      ^

TypeError: net.createTrainStream is not a function
    at Object.<anonymous> (C:\Users\Username\Desktop\brain-tst\app.js:5:23)
    at Module._compile (internal/modules/cjs/loader.js:702:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:713:10)
    at Module.load (internal/modules/cjs/loader.js:612:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:551:12)
    at Function.Module._load (internal/modules/cjs/loader.js:543:3)
    at Function.Module.runMain (internal/modules/cjs/loader.js:744:10)
    at startup (internal/bootstrap/node.js:238:19)
    at bootstrapNodeJSCore (internal/bootstrap/node.js:572:3)
```

## *Where* does it happen?

It happens when trying to use createTrainStream() with LTSM network

## *How* do we replicate the issue?

Just try to train LTSM network

```
var brain = require('brain.js');
var net = new brain.recurrent.LSTM();
var trainStream = net.createTrainStream();
```
First error will be not existing function so we don't need to care about what's inside it. 

## *How* important is this (1-5)?

4

## Expected behavior (i.e. solution)

Stream Training is possible with LTSM network

## Other Comments

It's possible that I messed up something because I'm not familiar with Brain.js

**Comment by robertleeplummerjr** [comment]
This isn't currently supported, though we will have it implemented soon for LSTM.  Can you share your whole example with us, and we can use it as a use case?

**Comment by robertleeplummerjr** [comment]
Fyi, this is now supported in https://github.com/BrainJS/brain.js/pull/311

**Comment by ServOKio** [comment]
Now I'm facing the same problem. On version 1.6.1 this does not work so far, 2. 2.0.0-alpha.2 throws error "TypeError: Cannot set property 'outputConnector' of undefined"
(NPM)

**Comment by mubaidr** [comment]
This feature will be available in v2

**Comment by harrysayers** [comment]
Is there an update on this - I'm getting an error ('Not yet implemented') when using Train Stream with either RNN or LSTM.

**Comment by robertleeplummerjr** [comment]
I can take a look this week.

**Comment by harrysayers** [comment]
> I can take a look this week. 

Thank you 👍

**Comment by robertleeplummerjr** [comment]
There was an incomplete method within the `RNN` class.  I went ahead and finished it out, and published it in ([v2.0.0-beta.18](https://www.npmjs.com/package/brain.js/v/2.0.0-beta.18)).  I also needed to alter how the `TrainStream` class handled the net, and have it now working nicely with it, and published it as well: https://www.npmjs.com/package/train-stream/v/1.0.2.

The method `createTrainStream` on the `NeuralNetwork` class has been removed, because of compatibility issues with brain.js's typescript and browser implementations.  All the functionality is still there, but now it comes from `TrainStream`.

Thank you for your patience on this issue, and please let me know if you have any trouble.  Closing as now this issue is complete.
